grafana-plugin/src/containers/RotationForm/ShiftSwapForm.tsx

```diff
@@ -0,0 +1,211 @@
+import cn from 'classnames/bind';
+import dayjs from 'dayjs';
+import Draggable from 'react-draggable';
+
+import Modal from 'components/Modal/Modal';
+import Tag from 'components/Tag/Tag';
+import Text from 'components/Text/Text';
+import WithConfirm from 'components/WithConfirm/WithConfirm';
+import { WithPermissionControlTooltip } from 'containers/WithPermissionControl/WithPermissionControlTooltip';
+import { SHIFT_SWAP_COLOR } from 'models/schedule/schedule.helpers';
+import { Schedule, ShiftSwap } from 'models/schedule/schedule.types';
+import { getTzOffsetString } from 'models/timezone/timezone.helpers';
+import { Timezone } from 'models/timezone/timezone.types';
+import { getUTCString } from 'pages/schedule/Schedule.helpers';
+import { useStore } from 'state/useStore';
+import { UserActions } from 'utils/authorization';
+
+import DateTimePicker from './parts/DateTimePicker';
+import UserItem from './parts/UserItem';
+
+import styles from './RotationForm.module.css';
+
+const cx = cn.bind(styles);
+
+interface ShiftSwapFormProps {
+  id: ShiftSwap['id'] | 'new';
+  scheduleId: Schedule['id'];
+  params: Partial<ShiftSwap>;
+  currentTimezone: Timezone;
+
+  onUpdate: () => void;
+
+  onHide: () => void;
+}
+
+const ShiftSwapForm = (props: ShiftSwapFormProps) => {
+  const { onUpdate, onHide, id, scheduleId, params: defaultParams, currentTimezone } = props;
+
+  const [shiftSwap, setShiftSwap] = useState({ ...defaultParams });
+
+  const store = useStore();
+  const { scheduleStore } = store;
+
+  useEffect(() => {
+    if (id !== 'new') {
+      scheduleStore.loadShiftSwap(id).then(setShiftSwap);
+    }
+  }, [id]);
+
+  useEffect(() => {
+    if (defaultParams) {
+      setShiftSwap({ ...shiftSwap, swap_start: defaultParams.swap_start, swap_end: defaultParams.swap_end });
+    }
+  }, [defaultParams]);
+
+  const handleShiftSwapStartChange = useCallback(
+    (value) => {
+      setShiftSwap({ ...shiftSwap, swap_start: getUTCString(value) });
+    },
+    [shiftSwap]
+  );
+
+  const handleShiftSwapEndChange = useCallback(
+    (value) => {
+      setShiftSwap({ ...shiftSwap, swap_end: getUTCString(value) });
+    },
+    [shiftSwap]
+  );
+
+  const handleDescriptionChange = useCallback(
+    (event) => {
+      setShiftSwap({ ...shiftSwap, description: event.target.value });
+    },
+    [shiftSwap]
+  );
+
+  const handleCreate = useCallback(async () => {
+    await scheduleStore.createShiftSwap({ schedule: scheduleId, ...shiftSwap });
+
+    onHide();
+    onUpdate();
+  }, [shiftSwap]);
+
+  const handleDelete = useCallback(async () => {
+    await scheduleStore.deleteShiftSwap(id);
+
+    onHide();
+    onUpdate();
+  }, [id]);
+
+  const handleTake = useCallback(async () => {
+    await scheduleStore.takeShiftSwap(id);
+
+    onHide();
+    onUpdate();
+  }, [id]);
+
+  const beneficiaryName = shiftSwap?.beneficiary && store.userStore.items[shiftSwap.beneficiary]?.name;
+
+  const readOnly = id !== 'new';
```

Comment:
could you please name this one `isReadOnly` instead? booleans with `is` are easier to read.